public/js/project/utilities/math.js

```diff
@@ -68,6 +68,20 @@ module.exports = {
 		}
 		return validPoints;
 	},
+	getValidPreImageOval: function(matrix) {
```

Comment:
Again, validating doesn't hurt, but isn't necessary since we are choosing the values.